src/inventory/ArmorInventory.php

```diff
@@ -36,6 +40,14 @@ public function __construct(
 		protected Living $holder
 	){
 		parent::__construct(4);
+
+		$this->validators->add(static function (Inventory $inventory, Item $item, int $slot) : bool{
+			return ($item instanceof Armor && $item->getArmorSlot() === $slot) ||
+				($slot === ArmorInventory::SLOT_HEAD && $item instanceof ItemBlock && (
+						$item->getBlock()->getTypeId() === BlockTypeIds::CARVED_PUMPKIN ||
+						$item->getBlock() instanceof MobHead
```

Comment:
I'd prefer checking type ID or tags here
Instanceof is rarely the right solution for this